src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -583,9 +693,294 @@ bool MatchesClientId(Task<IIdentity> identity)
             }
         }
 
+        private async Task<Vault> ValidateAndGetExistingKeyVault()
+        {
+            if (string.IsNullOrWhiteSpace(configuration.KeyVaultName))
+            {
+                return null;
+            }
+
+            return (await GetKeyVaultAsync(configuration.KeyVaultName))
+                ?? throw new ValidationException($"If key vault name is provided, it must already exist in region {configuration.RegionName}, and be accessible to the current user.", displayExample: false);
+        }
+
+        private async Task<FlexibleServerModel.Server> ValidateAndGetExistingPostgresqlServer()
+        {
+            if (string.IsNullOrWhiteSpace(configuration.PostgreSqlServerName))
+            {
+                return null;
+            }
+
+            return (await GetExistingPostgresqlService(configuration.PostgreSqlServerName))
+                ?? throw new ValidationException($"If Postgresql server name is provided, the server must already exist in region {configuration.RegionName}, and be accessible to the current user.", displayExample: false);
+        }
+
+        private async Task<ManagedCluster> ValidateAndGetExistingAKSClusterAsync()
+        {
+            if (string.IsNullOrWhiteSpace(configuration.AksClusterName))
+            {
+                return null;
+            }
+
+            return (await GetExistingAKSClusterAsync(configuration.AksClusterName))
+                ?? throw new ValidationException($"If AKS cluster name is provided, the cluster must already exist in region {configuration.RegionName}, and be accessible to the current user.", displayExample: false);
+        }
+        
+        private async Task<FlexibleServerModel.Server> GetExistingPostgresqlService(string serverName)
+        {
+            return (await Task.WhenAll(subscriptionIds.Select(async s =>
+            {
+                try
+                {
+                    var client = new FlexibleServer.PostgreSQLManagementClient(tokenCredentials) { SubscriptionId = s };
+                    return await client.Servers.ListAsync();
+                }
+                catch (Exception)
```

Comment:
I'm not sure which specific exception we need to catch actually, I was following the example of getting the existing batch account. In this specific case, inside a linq query, it might be okay to catch all exceptions and return the empty list / throw the validationexception at the end?

Reply:
I think that's good, we should at least write the exception to the console though before returning null

Reply:
Fixed